src/web/crate_details.rs

```diff
@@ -1077,42 +1236,142 @@ mod tests {
 
     #[test]
     fn platform_links_are_direct_and_without_nofollow() {
+        fn check_links(response_text: String, ajax: bool, should_contain_redirect: bool) {
+            let platform_links: Vec<(String, String)> = kuchikiki::parse_html()
+                .one(response_text)
+                .select(&format!(r#"{}li a"#, if ajax { "" } else { "#platforms " }))
+                .expect("invalid selector")
+                .map(|el| {
+                    let attributes = el.attributes.borrow();
+                    let url = attributes.get("href").expect("href").to_string();
+                    let rel = attributes.get("rel").unwrap_or("").to_string();
+                    (url, rel)
+                })
+                .collect();
+
+            assert_eq!(platform_links.len(), 2);
+
+            for (url, rel) in platform_links {
+                assert_eq!(
+                    url.contains("/target-redirect/"),
+                    should_contain_redirect,
+                    "ajax: {ajax:?}, should_contain_redirect: {should_contain_redirect:?}",
+                );
+                if !should_contain_redirect {
+                    assert_eq!(rel, "");
+                } else {
+                    assert_eq!(rel, "nofollow");
+                }
+            }
+        }
+
         wrapper(|env| {
             env.fake_release()
                 .name("dummy")
                 .version("0.4.0")
                 .rustdoc_file("dummy/index.html")
                 .rustdoc_file("x86_64-pc-windows-msvc/dummy/index.html")
+                .rustdoc_file("x86_64-pc-windows-msvc/dummy/struct.A.html")
                 .default_target("x86_64-unknown-linux-gnu")
                 .add_target("x86_64-pc-windows-msvc")
                 .create()?;
 
             let response = env.frontend().get("/crate/dummy/0.4.0").send()?;
             assert!(response.status().is_success());
+            check_links(response.text()?, false, false);
+            // Same test with AJAX endpoint.
+            let response = env
+                .frontend()
+                .get("/-/menus/platforms/crate/dummy/0.4.0")
```

Comment:
I added the test but it fails because:

```
Content-Security-Policy: The page’s settings blocked the loading of a resource at http://localhost:3000/-/menus/platforms/crate/sysinfo/0.29.7 (“default-src”).
```

So I took a look and I think that the missing `nonce` attribute on the script is what's preventing it from making AJAX queries. However, `csp_nonce` doesn't seem to be generated for `/crate/:name/` and `/crate/:name/:version` pages. Is there a way to make it generate it if needed?

Reply:
Currently rustdoc pages are excluded from CSP, from what I remember because there is a risk of docs breaking that bring their own JS.

Reply:
You need to add `connect-src 'self'` to the policy generated in `src/web/csp.rs` to allow the XHR. The `nonce` is generated and applied to the script, but that doesn't affect what XHR's are allowed from the script, those hit the `default-src 'none'` policy.

Reply:
(Though, that then hits a 404 because the generated URL needs some adjustments).

Reply:
That was it, thanks a lot! I'll update the route as well to allow this one.